### PR TITLE
Updated datasets.py

### PR DESCRIPTION
**Pull request by ShreyParikh07**
Datasets implementation

### CODE REVIEW

**Comment by eroell** [comment]
thanks for the PR 🥳 comments in the files!

File: docs/notebooks/ehrapy_data/GIBleed_dataset/GiBleed_5.3/CARE_SITE.csv [hunk 1]
```diff
@@ -0,0 +1 @@
+CARE_SITE_ID,CARE_SITE_NAME,PLACE_OF_SERVICE_CONCEPT_ID,LOCATION_ID,CARE_SITE_SOURCE_VALUE,PLACE_OF_SERVICE_SOURCE_VALUE
```
Comment (eroell) [inline docs/notebooks/ehrapy_data/GIBleed_dataset/GiBleed_5.3/CARE_SITE.csv:1]: no csv files stored here, rather download from external :)

File: src/ehrdata/dt/datasets.py [hunk 2]
```diff
@@ -62,7 +62,7 @@ def mimic_iv_omop(backend_handle: DuckDBPyConnection, data_path: Path | None = N
         >>> con.execute("SHOW TABLES;").fetchall()
     """
     if data_path is None:
-        data_path = "ehrapy_data/mimic-iv-demo-data-in-the-omop-common-data-model-0.9"
+        data_path = Path("ehrapy_data/mimic-iv-demo-data-in-the-omop-common-data-model-0.9")
```
Comment (eroell) [inline src/ehrdata/dt/datasets.py:65]: yep good one

**Comment by eroell** [comment]
And add the 3 datasets in pytest:
you can e.g. just call `ep.dt.<dataset X>` in a `function test_loading_datasets.py` script within a function `test_loading_dataset()`, that gets called by pytest and checks it actually downloads the desired data; they are small enough I believe for this to be done

File: src/ehrdata/dt/datasets.py [hunk 3]
```diff
@@ -81,7 +81,10 @@ def mimic_iv_omop(backend_handle: DuckDBPyConnection, data_path: Path | None = N
...
             return
 
-    return _set_up_duckdb(data_path + "/1_omop_data_csv", backend_handle)
+    extracted_folder = next(
+        (folder for folder in data_path.iterdir() if folder.is_dir() and "_csv" in folder.name), data_path
+    )
+    return _set_up_duckdb(extracted_folder, backend_handle)
 
 
 def gibleed_omop(backend_handle: DuckDBPyConnection, data_path: Path | None = None) -> None:
```
Comment (eroell) [inline src/ehrdata/dt/datasets.py:90]: Adding more information to the docstrings would be great; see e.g. other ehrapy datasets that we already have, with links to where the data is from, and that this is in the omop data format with maybe a link to the omop data description

**Review by eroell — CHANGES_REQUESTED** [review]

File: tests/test_datasets.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+@pytest.fixture(scope="session", autouse=True)
+def cleanup_test_data():
+    """Fixture to clean up test data directory after tests."""
+    yield
+    if TEST_DATA_DIR.exists():
+        shutil.rmtree(TEST_DATA_DIR)
+
+
+if __name__ == "__main__":
```
Comment (Zethson) [inline tests/test_datasets.py:61]: Don't think that this is necessary?

File: src/ehrdata/dt/datasets.py [hunk 5]
```diff
@@ -137,9 +170,39 @@ def synthea27nj_omop(backend_handle: DuckDBPyConnection, data_path: Path | None
         >>> ed.dt.synthea27nj_omop(backend_handle=con)
         >>> con.execute("SHOW TABLES;").fetchall()
     """
-    # TODO
-    # https://github.com/darwin-eu/EunomiaDatasets/tree/main/datasets/Synthea27Nj
-    raise NotImplementedError()
+    if data_path is None:
+        data_path = Path("ehrapy_data/Synthea27Nj")
+
+    if data_path.exists():
```
Comment (Zethson) [inline src/ehrdata/dt/datasets.py:176]: I'd have a general downloading function that you can reuse. Just steal the one from ehrapy. Consider not printing but use proper logging.

**Review by Zethson — COMMENTED** [review]
Just random things I saw

**Comment by eroell** [comment]
Some things here became part of #51 and #62, but as these branches have diverged quite a bit likely will close this PR soon

File: tests/test_datasets.py (hunk 4 above)
Comment (eroell) [reply tests/test_datasets.py:61]: don't think so either, thx for the looking over, appreciate!

**Pull request closed without merging**